src/by/string/regex/Task1.java

```diff
@@ -0,0 +1,187 @@
+                for (int j = 1; j < numberOfOccurrences.length; j++) {
+                    hasEqualOccurrences = hasEqualOccurrences && firstNumberOfOccurrences == numberOfOccurrences[j];
+                }
+                if (hasEqualOccurrences) {
+                    sortAbc(words);
+                } else {
+                    sort321(words, numberOfOccurrences);
+                }
+                sentences[i1] = joinStrings(words);
+            }
+            paragraphs[i] = joinStrings(sentences);
+            System.out.println(paragraphs[i]);
+        }
+    }
+
+    static String joinStrings(String [] arrayOfStrings) {
+        StringJoiner sj = new StringJoiner(" ");
+        for (String arrayOfString : arrayOfStrings) {
+            sj.add(arrayOfString);
+        }
+        return sj.toString();
+    }
+
+    static String [] splitIntoParagraphs(String str) {
+        return str.split("\\n");
+    }
+
+    static String [] splitIntoSentences(String paragraph) {
+        return paragraph.split("[\\.!\\?]");
+    }
+
+    static String [] splitIntoWords(String sentence) {
+        return sentence.split("\\s");
+    }
+
+    static int [] findOutTheLength(String [] a) {
+        int [] length = new int[a.length];
+        for (int i = 0; i < a.length; i++) {
+            length[i] = a[i].length();
+        }
+        return length;
+    }
+
+    static String [] sort123(String [] arrayOfStrings, int [] numbers) {
+        for (int i = 0; i < numbers.length; i++) {
+            for (int j = 0; j < numbers.length; j++) {
+                if (numbers[i] < numbers[j]) {
+                    String tempString = arrayOfStrings[i];
+                    arrayOfStrings[i] = arrayOfStrings[j];
+                    arrayOfStrings[j] = tempString;
+                    int tempInt = numbers[i];
+                    numbers[i] = numbers[j];
+                    numbers[j] = tempInt;
+                }
+            }
+        }
+        return arrayOfStrings;
+    }
+
+    static void sort321(String [] arrayOfStrings, int [] numbers) {
+        for (int i = 0; i < numbers.length; i++) {
+            for (int j = 0; j < numbers.length; j++) {
+                if (numbers[i] > numbers[j]) {
+                    String tempString = arrayOfStrings[i];
+                    arrayOfStrings[i] = arrayOfStrings[j];
+                    arrayOfStrings[j] = tempString;
+                    int tempInt = numbers[i];
+                    numbers[i] = numbers[j];
+                    numbers[j] = tempInt;
+                }
+            }
+        }
+    }
+
+    static void sortAbc(String [] arrayOfStrings) {
+        String tempString;
+        for (int i = 0; i < arrayOfStrings.length; i++) {
+            for (int j = i + 1; j < arrayOfStrings.length; j++) {
+                if (arrayOfStrings[i].compareToIgnoreCase(arrayOfStrings[j]) > 0) {
+                    tempString = arrayOfStrings[i];
```

Comment:
I'd introduce this method because you use swap operation in many places so there would be less code duplication
```
public static final <T> void swap (T[] a, int i, int j) {
  T t = a[i];
  a[i] = a[j];
  a[j] = t;
}
```